src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1957,6 +1959,21 @@ MavlinkReceiver::handle_message_heartbeat(mavlink_message_t *msg)
 			tstatus.system_id = msg->sysid;
 			tstatus.component_id = msg->compid;
 		}
+
+		/* handle avoidance heartbeats */
+		if (hb.type == MAV_TYPE_SUBMARINE) {  //missuse submarine type for avoidance
```

Comment:
I think we need to figure out how to set component id in mavros: https://github.com/mavlink/mavros/pull/1118